modules/collect/src/main/java/com/opengamma/strata/collect/Messages.java

```diff
@@ -110,4 +120,57 @@ public static String format(String messageTemplate, Object... args) {
     return builder.toString();
   }
 
+  /**
+   * Formats a templated message inserting named arguments. Typical template would look like "Hello, {attributeName}".
```

Comment:
I still think this needs an example. I don't think it's clear from the docs what it does. Something like:

    Messages.formatWithAttributes("Foo={foo}, Bar={}", "abc", 123);

returns the message 

    "Foo=abc, Bar=123" 

and the map 

    {"foo": "123"}

Feel free to copy and paste :)